rcore/overlay_manager.c

```diff
@@ -0,0 +1,218 @@
+/* overlay_manager.c
+ * Routines for managing the overlay window and it's content
+ * RebbleOS
+ * 
+ * Author: Barry Carter <barry.carter@gmail.com>.
+ */
+#include "overlay_manager.h"
+
+/* A message to talk to the overlay thread */
+typedef struct OverlayMessage {
+    uint8_t command;
+    void *data;
+} OverlayMessage;
+
+#define OVERLAY_CREATE    0
+#define OVERLAY_DRAW      1
+#define OVERLAY_DESTROY   2
+
+static xQueueHandle _overlay_queue;
+static void _overlay_thread(void *pvParameters);
+static list_head _overlay_window_list_head = LIST_HEAD(_overlay_window_list_head);
+static void _overlay_window_draw(void);
+
+void overlay_window_init(void)
+{
+    _overlay_queue = xQueueCreate(1, sizeof(struct OverlayMessage));
+    
+    app_running_thread *thread = appmanager_get_thread(AppThreadOverlay);
+    thread->status = AppThreadLoaded;
+    thread->thread_entry = &_overlay_thread;
+    /* start the thread 
+     * We are only using the thread launcher in appmanager for this
+     * not the full supervisory process. It's lightweight
+     */
+    appmanager_execute_app(thread, 0);    
+}
+
+/*
+ * Create a new top level window and all of the contents therein
+ */
+void overlay_window_create(OverlayCreateCallback creation_callback)
+{
+    OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_CREATE,
+        .data = (void *)creation_callback
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+void overlay_window_draw(void)
+{
+    OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_DRAW,
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+
+void overlay_window_destroy(OverlayWindow *overlay_window)
+{
+        OverlayMessage om = (OverlayMessage) {
+        .command = OVERLAY_DESTROY,
+        .data = (void *)overlay_window
+    };
+    xQueueSendToBack(_overlay_queue, &om, 0);
+}
+
+list_head *overlay_window_thread_get_head(Window *window)
```

Comment:
I admit in my bleary-eyed state that I do not understand the intent behind this function.  Perhaps a comment?